### DISCUSSION TITLE
Binary Failed to download on ZorinOS

### DISCUSSION THREAD

**Discussion by muhammadyusuf-kurbonov** [typegoose/mongodb-memory-server#595] [Q&A]
## Versions

- NodeJS: 16.13.1
- mongodb-memory-server-*: latest
- mongodb: 5.0.5
- mongoose: 7.3.1 
- system: Linux - Zorin OS based on Ubuntu focal

package: mongo-memory-server 

What am I doing wrong?

```
error: failed to connect to server [localhost:27017] on first connect [Error: connect ECONNREFUSED 127.0.0.1:27017
    at TCPConnectWrap.afterConnect [as oncomplete] (node:net:1161:16) {
  name: 'MongoNetworkError',
  errorLabels: [Array],
  [Symbol(mongoErrorContextSymbol)]: {}
}] {"name":"MongoNetworkError","errorLabels":["TransientTransactionError"]}
Couldnt download "https://fastdl.mongodb.org/linux/mongodb-linux-x86_64-ubuntu1604-5.0.5.tgz"! connect ECONNREFUSED 127.0.0.1:443
Starting the instance failed, enable debug for more information
[bugsnag] Unhandled rejection…
Error: connect ECONNREFUSED 127.0.0.1:443
    at TCPConnectWrap.afterConnect [as oncomplete] (node:net:1161:16)
[bugsnag] Unhandled rejection…
MongoNetworkError: failed to connect to server [localhost:27017] on first connect [Error: connect ECONNREFUSED 127.0.0.1:27017
    at TCPConnectWrap.afterConnect [as oncomplete] (node:net:1161:16) {
  name: 'MongoNetworkError'
}]
```

code is

```ts
mongoServer = await MongoMemoryServer.create({
            binary: {
                version: '5.0.5',
                skipMD5: true,
            },
        });
        mongoServer.getConnectionString().then(console.log);
```

**Comment by hasezoey** [comment]
next time please use triple (```) for code blocks

---

from what i can tell, you are trying to connect to something that is not even started

---

> Couldnt download "https://fastdl.mongodb.org/linux/mongodb-linux-x86_64-ubuntu1604-5.0.5.tgz"! connect ECONNREFUSED 127.0.0.1:443

it seems like it could not download a binary for your system, if `ZorinOS` is based on ubuntu, but does not have a (or a wrong) `/etc/upstream/os-release` file, so you would have to manually define the binary to use, in your case it would probably be `ARCHIVE_NAME=mongodb-linux-x86_64-ubuntu2004-5.0.5.tgz`

> mongodb-memory-server-*: latest

`latest` is not a version, please use actual versions

---

PR's for support for `ZorinOS` are welcome, follow the format like `Linux Mint` and `Elementary` do: https://github.com/nodkz/mongodb-memory-server/blob/afd8c5eb8667bedab47717e6ea0bf5e8080340d9/packages/mongodb-memory-server-core/src/util/MongoBinaryDownloadUrl.ts#L350-L375

---

PS: changed title to reflect issue
